Runtime/Services/BacktraceSession.cs

```diff
@@ -0,0 +1,317 @@
+﻿using Backtrace.Unity.Common;
+using Backtrace.Unity.Interfaces;
+using Backtrace.Unity.Json;
+using Backtrace.Unity.Model;
+using Backtrace.Unity.Model.JsonData;
+using Backtrace.Unity.Model.Session;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Backtrace.Unity.Tests.Runtime")]
+namespace Backtrace.Unity.Services
+{
+    internal sealed class BacktraceSession : IBacktraceSession
+    {
+        /// <summary>
+        /// Default number of retries
+        /// </summary>
+        public const int DefaultNumberOfRetries = 3;
+
+        /// <summary>
+        /// Submission url
+        /// </summary>
+        public string SubmissionUrl { get; set; }
+
+        /// <summary>
+        /// Determine if http client should ignore ssl validation
+        /// </summary>
+        public bool IgnoreSslValidation
+        {
+            get
+            {
+                return RequestHandler.IgnoreSslValidation;
+            }
+            set
+            {
+                RequestHandler.IgnoreSslValidation = value;
+            }
+        }
+
+        private int _numberOfDroppedEvents = 0;
+
+        /// <summary>
+        /// List of unique events that will be added to next session submission payload
+        /// </summary>
+        internal readonly LinkedList<UniqueEvent> UniqueEvents = new LinkedList<UniqueEvent>();

```

Comment:
In practice, it is likely that using whatever C#'s equivalent of a vector or array is will perform slightly better, especially if initial capacity is reserved.

Reply:
I'm using LinkedList because I want to add events to the beginning of the queue when http error occurred and we want to retry http request. Normally rest of the events go to the end.